File: 프로그래머스/3/12938. 최고의 집합/최고의 집합.java
import java.util.Arrays;

class Solution {
	public int[] solution(int n, int s) {
		int i;
		int num;
		int thr;
		int[] ans;

		if (n > s) {
			return new int[] {-1};
		}
		ans = new int[n];
		num = (s - 1) / n;
		thr = n - ((s - 1) % n + 1);
		for (i = 0; i < thr; i++) {
			ans[i] = num;
		}
		num++;
		for (; i < n; i++) {
			ans[i] = num;
		}
		return ans;
	}
}

File: 프로그래머스/3/12938. 최고의 집합/최고의 집합.cpp
#include <string>
#include <vector>

using namespace std;

constexpr int FAIL = -1;

vector<int> solution(int n, int s) {
    int i;
    int num;
    int thr;
    vector<int> ans;

    if (n > s) {
        ans.emplace_back(FAIL);
        return ans;
    }
    ans.reserve(n);
    num = (s - 1) / n;
    thr = n - ((s - 1) % n + 1);
    for (i = 0; i < thr; i++) {
        ans.emplace_back(num);
    }
    num++;
    for (; i < n; i++) {
        ans.emplace_back(num);
    }
    return ans;
}